search/sublist_search.cpp

```diff
@@ -0,0 +1,338 @@
+        log("Assertion check passed!");
+
+        log("[PASS] : TEST CASE 1 PASS!");
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+
+        delete (sublistLL);
+        delete (mainlistLL);
+    }
+
+    /**
+     * @brief A test case which contains main list of 100 elements and sublist
+     * of 20.
+     * @returns void
+     * */
+    void testCase_2() {
+        const bool expectedOutput = true;
+
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+        log("This is test case 2 for sublist search Algorithm : ");
+        log("Description:");
+        log("   contains main list of 100 elements and sublist of 20");
+
+        std::vector<uint64_t> sublistData(20);
+        std::vector<uint64_t> mainlistData(100);
+
+        for (int i = 0; i < 100; i++) {
+            // Inserts 100 elements in main list
+            mainlistData[i] = i + 1;
+        }
+
+        int temp = 0;
+        for (int i = 45; i < 65; i++) {
+            // Inserts 20 elements in sublist
+            sublistData[temp] = i + 1;
+            temp++;
+        }
+
+        search::sublist_search::Node *sublistLL =
+            search::sublist_search::makeLinkedList(sublistData);
+        search::sublist_search::Node *mainlistLL =
+            search::sublist_search::makeLinkedList(mainlistData);
+
+        bool exists =
+            search::sublist_search::sublistSearch(sublistLL, mainlistLL);
+
+        log("Checking assert expression...");
+        assert(exists == expectedOutput);
+        log("Assertion check passed!");
+
+        log("[PASS] : TEST CASE 2 PASS!");
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+    }
+
+    /**
+     * @brief A test case which contains main list of 50 elements and sublist
+     * of 20.
+     * @returns void
+     * */
+    void testCase_3() {
+        const bool expectedOutput = false;
+
+        log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+            "~");
+        log("This is test case 3 for sublist search Algorithm : ");
+        log("Description:");
+        log("   contains main list of 50 elements and sublist of 20");
+
+        std::vector<uint64_t> sublistData(20);
+        std::vector<uint64_t> mainlistData(50);
+
+        for (int i = 0; i < 50; i++) {
+            // Inserts 100 elements in main list
+            mainlistData.push_back(i + 1);
+        }
+
+        for (int i = 45; i < 65; i++) {
+            // Inserts 20 elements in sublist
+            sublistData.push_back(i + 1);
+        }
+
+        search::sublist_search::Node *sublistLL =
+            search::sublist_search::makeLinkedList(sublistData);
+        search::sublist_search::Node *mainlistLL =
+            search::sublist_search::makeLinkedList(mainlistData);
+
+        bool exists =
+            search::sublist_search::sublistSearch(sublistLL, mainlistLL);
```

Comment:
Add a one-line description of what the variable is for (see the example below). Check other parts of the code, too.
```c++
int64_t data = 0;       ///< The value/key of the node.
```